src/Libraries/DSCPython/Encoders/BigIntegerEncoder.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Numerics;
+using Python.Runtime;
+
+namespace DSCPython.Encoders
+{
+    internal class BigIntegerEncoder : IPyObjectEncoder, IPyObjectDecoder
+    {
+        public bool CanDecode(PyObject objectType, Type targetType)
+        {
+            return targetType == typeof(BigInteger);
+        }
+
+        public bool CanEncode(Type type)
+        {
+            return type == typeof(BigInteger);
+        }
+
+        public bool TryDecode<T>(PyObject pyObj, out T value)
+        {
+            if (!PyLong.IsLongType(pyObj))
+            {
+                value = default;
+                return false;
+            }
+
+            using (var pyLong = PyLong.AsLong(pyObj))
+            {
+                value = (T)(object)pyLong.ToBigInteger();
```

Comment:
staring at this I'm assuming that PythonNet will only ever call this method when T is `bigInteger` ?

Reply:
Yes. I think there is something strange with the definition of the interface because it's not a generic one but one of its methods is generic. I didn't know how to handle that exactly but I took this from an example they have in a test.